research-software-directory/RSD-as-a-service
# Extended unit test coverage from 20% to 70%

# Increase unit tests coverage

Closes #675 

Changes proposed in this pull request:
* Write more unit test to increase test coverage 
* Removed unused components and methods
* Extended frontend/README test section with test scripts and when to use
* Added afterEach in jest.setup.js to manually call garbage collection. This is applicable to yarn test:memory script only.

How to test:
* examine the test coverage report of this PR
* or run unit tests locally in the frontend folder:
  *  `yarn install`: to install all dependencies
  * `yarn test:coverage` to run test with coverage report
* and run e2e test from the root with `make e2e-tests`
* in case of memory overflow errors you can run:
   *  `yarn test:memlimit`: run tests with limited memory use (10% of total and max of 2 workers)
   *  `yarn test:memory`: to see heap memory usage per test. When memory heap grows per test executation it indicates possible memory leak.
 
## Memory usage and additional scripts
There is change in memory management in node v18 and v16. This causes memory leak when running unit tests in Jest as noted by Ewan in the comments below. In some cases the PC can crash with memory overflow. For this cases additional test scripts are created: **yarn test:memlimit** and **yarn test:memory**.

## Printscreen unit test report

![image](https://user-images.githubusercontent.com/9204081/211533921-881e9664-9739-49fe-9432-44df30abaa2e.png)


PR Checklist:
